src/Compilers/CSharp/Portable/Symbols/Source/SourceNamespaceSymbol.cs

```diff
@@ -500,28 +500,31 @@ public override bool IsDefinedInSourceTree(SyntaxTree tree, TextSpan? definedWit
             return false;
         }
 
-        private readonly struct NameToSymbolMapBuilder
+        private readonly ref struct NameToSymbolMapBuilder
         {
-            private readonly Dictionary<string, object> _dictionary;
+            private readonly PooledDictionary<string, object> _dictionary = PooledDictionary<string, object>.GetInstance();
 
-            public NameToSymbolMapBuilder(int capacity)
+            public NameToSymbolMapBuilder()
             {
-                _dictionary = new Dictionary<string, object>(capacity, StringOrdinalComparer.Instance);
+            }
+
+            public void Dispose()
+            {
+                _dictionary.Free();
             }
 
             public void Add(NamespaceOrTypeSymbol symbol)
             {
-                string name = symbol.Name;
-                object item;
-                if (_dictionary.TryGetValue(name, out item))
+                var name = symbol.Name;
+                if (_dictionary.TryGetValue(name, out var builderOrItem))
                 {
-                    var builder = item as ArrayBuilder<NamespaceOrTypeSymbol>;
-                    if (builder == null)
```

Comment:
>if (builder == null) [](http://example.com/codeflow?start=20&length=20)

Please restore original logic. It looks like there is no intent to change behavior here.

Reply:
Merged this all into common helper used by SourceMemberContainerSymbol as well.